File: static/archive/Competitive-Programming/UVA - 10943.py
def gcd(a,b):
    a=int(a)
    b=int(b)
    if b==0:
        return a
    else:
        return gcd(b,a%b)
while 1:
    n,k=map(int,input().split())
    if n==0 and k==0:
        break
    numerator=[]
    for i in range(k-1):
        numerator.append(int(n+i+1))
    for i in range(k-1):
        pivot=int(i+1)
        j=0
        while pivot!=1 and j<k-1:
            gcdd=gcd(pivot,numerator[j])
            if gcdd!=1:
                pivot/=gcdd
                numerator[j]/=gcdd
            j+=1
    ans=1
    for i in range(k-1):
        ans*=numerator[i]
        ans%=1000000
    print(int(ans))

File: static/archive/Competitive-Programming/UVA - 10943.cpp
#include <bits/stdc++.h>
using namespace std;
const long long MOD = 1000000;
// C(N+K-1,K-1)
long long N, K, Ans, GCD;
vector<long long> Numerator;
int main() {
  while (cin >> N >> K) {
    if (N == 0 && K == 0)
      return 0;
    Ans = 1;
    Numerator.clear();
    for (int i = 1; i <= K - 1; ++i)
      Numerator.push_back(N + i);
    for (int i = 1; i <= K - 1; ++i) {
      long long Pivot = i;
      int j = 0;
      while (Pivot != 1) {
        GCD = __gcd(Pivot, Numerator[j]);
        if (GCD != 1) {
          Pivot /= GCD;
          Numerator[j] /= GCD;
        }
        j++;
      }
    }
    for (auto i : Numerator) {
      Ans *= i;
      Ans %= MOD;
    }
    cout << Ans << endl;
  }
}